include/miopen/miopen.h

```diff
@@ -580,6 +598,22 @@ MIOPEN_EXPORT miopenStatus_t miopenCreateTensorDescriptor(miopenTensorDescriptor
 MIOPEN_EXPORT miopenStatus_t miopenSet4dTensorDescriptor(
     miopenTensorDescriptor_t tensorDesc, miopenDataType_t dataType, int n, int c, int h, int w);
 
+/*! @brief Set shape of ND tensor with specific layout
+ *
+ * Interface for setting N-D tensor shape. This interface support NHWC, NCHW, NCHWc*, CHWNc*
+ * @param tensorDesc   Tensor descriptor type (output)
+ * @param dataType     MIOpen datatype (input)
+ * @param tensorLayout Tensor layout (input)
```

Comment:
@pfultz2 Request for comment: I remember you are against forcing client giving the layout from API.

Reply:
@aska-0096 @carlushuang  for awareness

Reply:
@jerryyin @pfultz2 @aska-0096 @carlushuang @junliume Yes, IIRC we generally want to set strides via an API and let MIOpen do the rest of the work (derive layouts from them). So I'm also wondering why this approach isn't used here (due to complexity?)

Reply:
Well an enum is fine for an easy API fro some users as along as users can still set the strides as well to achieve the layout. The only way to do this correctly is not store a layout string or enum in the tensor descriptor.